events/src/deployment.rs

```diff
@@ -40,3 +40,34 @@ pub struct NodeLog {
     //pub source: String,
     pub msg: String,
 }
+
+#[derive(Serialize, Deserialize, Type, Debug, Clone)]
+pub struct NodeLogs {
+    pub logs: Vec<NodeLog>,
+}
+
+impl NodeLogs {
+    pub fn bytes_len(&self) -> usize {
+        self.logs.iter().fold(0, |len, log| len + log.msg.len())
```

Comment:
Nit: Easier to read as:

```suggestion
        self.logs.iter().map(|log| log.len()).sum()
```